public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -105,8 +105,12 @@ export class InspectJSONTab extends PureComponent<Props, State> {
           appEvents.emit(AppEvents.alertError, ['Unable to apply']);
         } else {
           const updates = JSON.parse(this.state.text);
+          const shouldUpdateGridPositionLayout = JSON.stringify(updates?.gridPos) !== JSON.stringify(panel?.gridPos);
           panel!.restoreModel(updates);
           panel!.refresh();
+          if (shouldUpdateGridPositionLayout) {
+            panel!.updateGridPositionLayout();
+          }
```

Comment:
Hmm, I wonder if this logic shouldn't belong to PanelModel? I mean maybe PanelModel should decide if it should publish the event or not. WDYT?

Reply:
That's a good point, so do you mean: to send the `updates` to a function in the `PanelModel`, and there we made the decision? right?

Reply:
Yes, exactly